src/main/java/org/openstreetmap/atlas/geography/atlas/change/ChangeAtlas.java

```diff
@@ -88,19 +100,26 @@ public Edge edge(final long identifier)
     @Override
     public Line line(final long identifier)
     {
-        throw new UnsupportedOperationException();
+        return entityFor(identifier, ItemType.LINE, () -> this.source.line(identifier),
+                (sourceEntity, overrideEntity) -> new ChangeLine(this, (Line) sourceEntity,
+                        (Line) overrideEntity));
     }
 
     @Override
     public Iterable<Line> lines()
     {
-        throw new UnsupportedOperationException();
+        return entitiesFor(ItemType.LINE, this::line, this.source.lines());
     }
 
     @Override
-    public AtlasMetaData metaData()
+    public synchronized AtlasMetaData metaData()
     {
-        throw new UnsupportedOperationException();
+        if (this.metaData == null)
+        {
+            final AtlasSize size = new AtlasSize(this);
+            this.metaData = this.source.metaData().copyWithNewSize(size).copyWithNewOriginal(false);
```

Comment:
What happens here if the source `metadata` is also `null`?

Reply:
NPE!!!

Reply:
Thinking about this some more, is it ever possible for `metaData` to actually be `null` in a valid `Atlas`? I don't think so. It looks like `PackedAtlasSerializer` will throw an exception if it cannot deserialize a field, so you'll never have a `null` metadata from a `PackedAtlas`, at least.

Reply:
I added a check anyway.